routes/auth.js

```diff
@@ -0,0 +1,64 @@
+const express = require('express')
+const bcrypt = require('bcryptjs')
+const jwt = require('jsonwebtoken')
+const User = require('../models/User')
+const auth = require('../middleware/auth')
+
+const router = express.Router()
+
+const { JWT_SECRET } = process.env
+
+router.post('/', async (request, response, next) => {
+  const { username, password } = request.body
+
+  try {
+    let user = await User.findOne({ username })
+
+    if (!user) {
+      const error = { message: 'Username does not exist!' }
+      return next({ status: 400, message: error.message })
+    }
+
+    bcrypt.compare(password, user.password, (errCompare, resCompare) => {
+      if (errCompare) {
+        console.error(errCompare)
+        return next({ status: 400, message: errCompare.message })
+      }
+      if (resCompare) {
+        const payload = {
+          user: {
+            id: user._id,
+          },
+        }
+
+        // TODO: expiration to 3600
```

Comment:
Do you need this comment?

Reply:
Yes, to remind me that later on I set the expiration to 3600, instead of 360000 ;)